pkg/tagger/collectors/workloadmeta_extract.go

```diff
@@ -121,6 +134,17 @@ func (c *WorkloadMetaCollector) processEvents(evBundle workloadmeta.EventBundle)
 				log.Errorf("cannot handle event for entity %q with kind %q", entityID.ID, entityID.Kind)
 			}
 
+			// remove the children seen in this iteration from the
+			// unseen list ...
+			for childTaggerID := range c.children[taggerEntityID] {
+				delete(unseen, childTaggerID)
+			}
+
+			// ... and remove entities for everything that has been
+			// left
+			source := fmt.Sprintf("%s-%s", workloadmetaCollectorName, string(entityID.Kind))
```

Comment:
Nit: this same `Sprintf` call is used in another function below. Would it make sense to extract this ?